src/plugins/aws/ssm/index.ts

```diff
@@ -266,6 +268,34 @@ const spawnSsmNode = async (
     });
   });
 
+/**
+ * Spawns a child process to add a private key to the ssh-agent. The SSH agent is included in the OpenSSH suite of tools
+ * and is used to hold private keys during a session. The SSH agent typically does not persist keys across system reboots
+ * or logout/login cycles. Once you log out or restart your system, any keys added to the SSH agent during that session
+ * will need to be added again in subsequent sessions.
+ */
+const executeScpCommand = async (
+  credential: AwsCredentials,
+  command: string,
+  privateKey: string
+) => {
+  // Execute should not leave any ssh-agent processes running after it's done performing an SCP transaction.
+  const execute = `
+  eval $(ssh-agent) >/dev/null 2>&1
+  trap 'kill $SSH_AGENT_PID' EXIT
+  ssh-add -q - <<< '${privateKey}' 
```

Comment:
Why is `<<<` important?

Reply:
It redirects the string as input. otherwise ssh-add interprets it as an option or a path incorrectly. Which is weird to me because we're still escaping things with a single quote.

> ssh-add: unrecognized option `-----BEGIN RSA PRIVATE KEY-----\r